### PR TITLE
feat: Fix for CSR reuse errors

### PR DESCRIPTION
**Pull request by manuvar-ms**
## Description

The default issuance policy using an integrated CA can cause pipeline errors of the form "CSR ... has already been used" on new versions of an existing certificate. This change adds a "reuseKey" parameter, which can be set to false to fix this error, according to https://learn.microsoft.com/en-us/azure/key-vault/certificates/create-certificate-signing-request?tabs=azure-portal. The default issue policy sets this to true, so the default value of the new parameter reflects that.

## Updating an existing module

- [x] I have run `brm validate` locally to verify the module files.
- [x] I have run deployment tests locally to ensure the module is deployable.
- [x] I have read the [Updating an existing module](https://github.com/Azure/bicep-registry-modules/blob/main/CONTRIBUTING.md#updating-an-existing-module) section in the contributing guide and updated the `version.json` file properly:
  - [x] The PR contains backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
- [x] I have updated the examples in README with the latest module version number.

### CODE REVIEW

**Review by dciborow — APPROVED** [review]

**Comment by shenglol** [comment]
https://github.com/Azure/bicep-registry-modules/pull/304 was merged which introduced some conflicts. Mind merging `main` and bumping the version again?

**Comment by dciborow** [comment]
lets first merge in this PR, then we will merge this in after #230

File: modules/deployment-scripts/create-kv-certificate/README.md [hunk 1]
```diff
@@ -55,7 +56,7 @@ param location string = resourceGroup().location
 param akvName string = 'yourAzureKeyVault'
 param certificateName string = 'myapp'
 
-module kvCert 'br/public:deployment-scripts/create-kv-certificate:3.2.1' = {
+module kvCert 'br/public:deployment-scripts/create-kv-certificate:3.3.1' = {
```
Comment (shenglol) [inline modules/deployment-scripts/create-kv-certificate/README.md:59]: Can you bump the version in the examples?

**Comment by dciborow** [comment]
merging this PR in with other PR with Key Vault updates

**Pull request closed without merging**